When modifying a PR, resolver fails on "Expected review comments to be initialized."

This workflow https://github.com/All-Hands-AI/openhands-resolver/actions/runs/11694555050/job/32568388367?pr=263

Encountered the following error:

```
--------
00:08:03 - openhands:INFO: resolve_issue.py:415 - Finished.
ERROR:root:  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 563, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 544, in main
    asyncio.run(
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 398, in resolve_issue
    output = await process_issue(
             ^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 244, in process_issue
    success, comment_success, success_explanation = issue_handler.guess_success(issue, state.history, llm_config)
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/issue_definitions.py", line 422, in guess_success
    raise ValueError("Expected review comments to be initialized.")

ERROR:root:<class 'ValueError'>: Expected review comments to be initialized.
Error: Process completed with exit code 1.
```

--- a/openhands_resolver/issue_definitions.py
+++ b/openhands_resolver/issue_definitions.py
@@ -119,6 +119,8 @@ class IssueHandler(IssueHandlerInterface):
             
             # Get issue thread comments
             thread_comments = self._get_issue_comments(issue["number"])
+            # Convert empty lists to None for optional fields
+            thread_comments = None if not thread_comments else thread_comments
             issue_details = GithubIssue(
                                 owner=self.owner,
                                 repo=self.repo,

--- a/openhands_resolver/issue_definitions.py
+++ b/openhands_resolver/issue_definitions.py
@@ -126,7 +126,7 @@ class IssueHandler(IssueHandlerInterface):
                                 title=issue["title"],
                                 body=issue["body"],
                                 thread_comments=thread_comments,
-                                review_comments=[],  # Initialize empty review comments for regular issues
+                                review_comments=None,  # Initialize review comments as None for regular issues
                             )
                 
             converted_issues.append(issue_details)

--- a/openhands_resolver/issue_definitions.py
+++ b/openhands_resolver/issue_definitions.py
@@ -75,7 +75,7 @@ class IssueHandler(IssueHandlerInterface):
         image_pattern = r'!\[.*?\]\((https?://[^\s)]+)\)'
         return re.findall(image_pattern, issue_body)
 
-    def _get_issue_comments(self, issue_number: int) -> list[str]:
+    def _get_issue_comments(self, issue_number: int) -> list[str] | None:
         """Download comments for a specific issue from Github."""
         url = f"https://api.github.com/repos/{self.owner}/{self.repo}/issues/{issue_number}/comments"
         headers = {
@@ -96,7 +96,7 @@ class IssueHandler(IssueHandlerInterface):
             all_comments.extend([comment["body"] for comment in comments])
             params["page"] += 1
 
-        return all_comments
+        return all_comments if all_comments else None
     
     def get_converted_issues(self) -> list[GithubIssue]:
         """Download issues from Github.
@@ -120,7 +120,6 @@ class IssueHandler(IssueHandlerInterface):
             # Get issue thread comments
             thread_comments = self._get_issue_comments(issue["number"])
             # Convert empty lists to None for optional fields
-            thread_comments = None if not thread_comments else thread_comments
             issue_details = GithubIssue(
                                 owner=self.owner,
                                 repo=self.repo,

--- a/openhands_resolver/issue_definitions.py
+++ b/openhands_resolver/issue_definitions.py
@@ -126,6 +126,7 @@ class IssueHandler(IssueHandlerInterface):
                                 title=issue["title"],
                                 body=issue["body"],
                                 thread_comments=thread_comments,
+                                review_comments=[],  # Initialize empty review comments for regular issues
                             )
                 
             converted_issues.append(issue_details)

--- a/tests/test_issue_handler.py
+++ b/tests/test_issue_handler.py
@@ -1,0 +1,41 @@
+import pytest
+from unittest.mock import patch, MagicMock
+from openhands_resolver.issue_definitions import IssueHandler
+from openhands_resolver.github_issue import GithubIssue
+
+def test_get_converted_issues_initializes_review_comments():
+    # Mock the necessary dependencies
+    with patch('requests.get') as mock_get:
+        # Mock the response for issues
+        mock_issues_response = MagicMock()
+        mock_issues_response.json.return_value = [{
+            'number': 1,
+            'title': 'Test Issue',
+            'body': 'Test Body'
+        }]
+        # Mock the response for comments
+        mock_comments_response = MagicMock()
+        mock_comments_response.json.return_value = []
+        
+        # Set up the mock to return different responses for different calls
+        # First call is for issues, second call is for comments
+        mock_get.side_effect = [mock_issues_response, mock_comments_response, mock_comments_response]  # Need two comment responses because we make two API calls
+        
+        # Create an instance of IssueHandler
+        handler = IssueHandler('test-owner', 'test-repo', 'test-token')
+        
+        # Get converted issues
+        issues = handler.get_converted_issues()
+        
+        # Verify that we got exactly one issue
+        assert len(issues) == 1
+        
+        # Verify that review_comments is initialized as an empty list
+        assert issues[0].review_comments == []
+        
+        # Verify other fields are set correctly
+        assert issues[0].number == 1
+        assert issues[0].title == 'Test Issue'
+        assert issues[0].body == 'Test Body'
+        assert issues[0].owner == 'test-owner'
+        assert issues[0].repo == 'test-repo'

--- a/tests/test_issue_handler.py
+++ b/tests/test_issue_handler.py
@@ -1,7 +1,5 @@
-import pytest
 from unittest.mock import patch, MagicMock
 from openhands_resolver.issue_definitions import IssueHandler
-from openhands_resolver.github_issue import GithubIssue
 
 def test_get_converted_issues_initializes_review_comments():
     # Mock the necessary dependencies
@@ -30,8 +28,8 @@ def test_get_converted_issues_initializes_review_comments():
         # Verify that we got exactly one issue
         assert len(issues) == 1
         
-        # Verify that review_comments is initialized as an empty list
-        assert issues[0].review_comments == []
+        # Verify that review_comments is initialized as None
+        assert issues[0].review_comments is None
         
         # Verify other fields are set correctly
         assert issues[0].number == 1

